### DISCUSSION TITLE
Redirecting subdomains to arbitrary location in- and outside project

### DISCUSSION THREAD

**Discussion by SeinopSys** [vercel/vercel#5622] [Ideas] [answered]
I would like to ask for a more intuitive solution to redirecting a subdomain connected to the project to an arbitrary URL of my choosing. There exists [this discussion](https://github.com/vercel/vercel/discussions/4051) but the solution there is to make a dummy project for what should essentially be 4 lines of configuration, just like [`redirects` in vercel.json](https://vercel.com/docs/configuration#project/redirects) but accepting domains as well.

My specific use case is having `stream.example.com` (where `example.com` is the project's primary domain) point to e.g. a [Twitch](https://twitch.tv) channel URL, this helps to link pages faster by having a more memorable URL. Yes, I could use a subpath and redirect that instead from within the project, but I specifically want to use subdomains because they are quicker and easier to type for me.

In my opinion this would most easily be achieved by amending the `redirects` config to support domains:

```json
{
  "redirects": [
    { "source": "//event.example.com/", "destination": "//example.com/about/event", "permanent": false },
    { "source": "//stream.example.com/", "destination": "https://twitch.tv/example", "permanent": false },
    { "source": "//stream.example.com/:match*", "destination": "https://twitch.tv/example", "permanent": false },
  ]
}
```

This shows a subdomain redirect that takes the user to a specific page on the primary domain, and another subdomain redirect that takes the user to an external site, just like a regular redirect. Both of these source domains would need to be verified in the project for the redirect to work to prevent abuse.

Bonus points if this could be done with a single entry but even if I have to duplicate it it's far less hassle than having to create separate Git repositories and next projects for a single domain redirect.

Alternatively, the Domains configuration page could have a freeform text input instead of a domain dropdown to enter the target URL instead.

**Comment by AlgusDark** [comment, marked as answer]
@SeinopSys , were you able to find a different solution from creating dummy projects? I was trying to use the `has` with `type` `host` but I had no success.

**Reply by SeinopSys** [reply]
No

**Reply by AlgusDark** [reply]
I was able to do it. My problem was that I was using "rewrites" instead of "redirects.

This configuration does the redirect for my subdomain https://git.algus.ninja
You just need to add a new domain in your project (in this case, git.algus.ninja)

```json
{
  "redirects": [
    {
      "source": "/",
      "has": [
        {
          "type": "host",
          "value": "git.algus.ninja"
        }
      ],
      "destination": "https://github.com/AlgusDark"
    }
  ]
}
```

I know this is almost 2 years after but it could help someone with the same problem.

**Reply by amyegan** [reply]
Thanks for sharing this answer Carlos! This is very similar to the solution recommended in the ["Can I redirect from a subdomain to a subpath?"](https://vercel.com/guides/can-i-redirect-from-a-subdomain-to-a-subpath) guide, and is the one I would most likely choose. Another option is to use DNS records as described in ["How do I point a subdomain to a service outside of Vercel?"](https://vercel.com/guides/pointing-subdomains-to-external-services)
